dart/test/monster_test_my_game.example2_generated.dart

```diff
@@ -26,6 +27,37 @@ class Monster {
   String toString() {
     return 'Monster{}';
   }
+
+  MonsterT unPack() => MonsterT();
+
+  static int pack(fb.Builder fbBuilder, MonsterT object) {
+    if (object == null) return 0;
+    return object.pack(fbBuilder);
+  }
+}
+
+class MonsterT {
+
+  int pack(fb.Builder fbBuilder) => _pack(fbBuilder, HashMap<Object, int>());
+
+  /// Write the object to the [Builder] only once and reuse the same offset on
+  /// subsequent references of the same object.
+  ///
+  /// Note that this method assumes you call it using the same [Builder]
+  /// instance every time. The returned offset is only good for the [Builder]
+  /// used in the first call to this method.
+  int packOnce(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    return existingOffsets[this] ??= _pack(fbBuilder, existingOffsets);
+  }
+
+  int _pack(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    assert(fbBuilder != null);
```

Comment:
this seems a bit superfluous in generated code.. most callers are other bits of generated code, and the user calling it with a `null` builder doesn't seem very likely.

Reply:
FWIW, these are stripped away in non-debug modes (and can be stripped away even in debug mode). 

However, if it's likely that this indicates an error in the generated code rather than in the caller, it'd be good to at least provide a message to that effect.  This also can go away whenever we get to migrate this code to NNBD, which I think someone is working on.

Reply:
Again as mentioned above - keeping in line with the previous `ObjectBuilder` code. Also, it doesn't hurt anything and is really only a temporary piece of code to be removed with null-safety.

That "someone" working on null-safety (or rather porting it from my work in the ObjectBox fork of flatbuffers) is me, at least I've said I would but didn't get to it yet. But it is on my list soon-ish, hopefully this PR will be merged by the time I do that, to avoid conflicts.